src/goto-harness/goto_harness_parse_options.h

```diff
@@ -9,12 +9,24 @@ Author: Diffblue Ltd.
 #ifndef CPROVER_GOTO_HARNESS_GOTO_HARNESS_PARSE_OPTIONS_H
 #define CPROVER_GOTO_HARNESS_GOTO_HARNESS_PARSE_OPTIONS_H
 
+#include <string>
+
+#include <goto-programs/goto_model.h>
 #include <util/parse_options.h>
 #include <util/ui_message.h>
 
-#define GOTO_HARNESS_OPTIONS "(version)" // end GOTO_HARNESS_OPTIONS
+#include "function_harness_generator_options.h"
+#include "goto_harness_generator_factory.h"
+
+// clang-format off
+#define GOTO_HARNESS_OPTIONS                                                   \
+  "(version)"                                                                  \
+  GOTO_HARNESS_FACTORY_OPTIONS                                                 \
+  FUNCTION_HARNESS_GENERATOR_OPTIONS                                           \
+// end GOTO_HARNESS_OPTIONS
+// clang-format on
 
-class goto_harness_parse_optionst : public parse_options_baset
+class goto_harness_parse_optionst : public parse_options_baset, public messaget
```

Comment:
If avoidable, I'd ask to use a class member instead of inheriting from `messaget`. (I know that's done elsewhere, but it's poor design. The relationship is "has a logger", not "is a logger.")

Reply:
:+1: sounds good to me